### ISSUE TITLE
Bug? Order of generic type parameters  `P` and `K` seems to be switched

### ISSUE THREAD

**Issue by Stef-Gijsberts** [fede1024/rust-rdkafka#105]
## Issue

I got a strange error while updating a codebase to use the new version of rust-rdkafka, 0.18.0.

This code:
```rust
let record = FutureRecord::to(&topic)
    .payload(&payload);

producer.send::<String, ()>(record, -1);
````

gave an error:
```
error[E0308]: mismatched types
   --> src/code.rs:116:45
    |
116 |                 producer.send::<String, ()>(record, -1);
    |                                             ^^^^^^ expected struct `std::string::String`, found ()
    |
    = note: expected type `rdkafka::producer::future_producer::FutureRecord<'_, std::string::String, ()>`
               found type `rdkafka::producer::future_producer::FutureRecord<'_, (), std::string::String>`
```

From looking at the docs, I would expect a `FutureRecord<'_, (), std::string::String>`, as the definition of `FutureRecord` is:

```rust
pub struct FutureRecord<'a, K: ToBytes + ?Sized + 'a, P: ToBytes + ?Sized + 'a>....
```

`K` comes first, then `P`.

## Suggested solution
**Maybe I am overlooking something and this is expected behaviour**, but in case it is not, here are my ideas on how to fix this bug.

### Option 1
Switch `P` and `K` on line `206` in `future_producer.rs`.

The original line:
https://github.com/fede1024/rust-rdkafka/blob/e152990256cc81d0b3d00146ef54d6f71ef1edf1/src/producer/future_producer.rs#L206

This would become:
```rust
 pub fn send<P, K>(&self, record: FutureRecord<K,P>, block_ms: i64) -> DeliveryFuture 
```

### Option 2 (I like this more)
Make the API consistent and change the definition of `FutureRecord`. First `P`, then `K`, just like the producers `send` function.

Thanks in advance!

**Comment by Stef-Gijsberts** [comment]
I think your solution is the best one :). It is the most consistent with the rest of the API.

**Comment by flavray** [comment]
I have pushed the fix to [`master`](https://github.com/fede1024/rust-rdkafka/commit/eb8d406da10a8f91293ddd1f61f94fdd4f77213f) and published `v0.19.0` to [crates.io](https://crates.io/crates/rdkafka/0.19.0) :)
